src/events.rs

```diff
@@ -31,7 +31,7 @@ pub struct NodeData {
 
 /// NotifyFn is used by the supervision API to send events to an interested
 /// listener.
-type NotifyFn = Box<dyn Fn(Event) -> BoxFuture<'static, ()>>;
+type NotifyFn = Box<dyn (Fn(Event) -> BoxFuture<'static, ()>) + Send + Sync>;
```

Comment:
Also, can you point out what prompted both the `Send` and `Sync` bounds?  I can understand `Send` being very commonly used that's necessary for being able to move things across threads which the runtime requires, but I don't think these require `Sync` which is a much more subtle trait bound, and actually substantially restrictive.  If this is because of the usage of `tokio::task::spawn` I believe the bound on that only requires `Send`.

Reply:
`tokio::spawn` is bending my arm

```
error[E0277]: `(dyn std::ops::Fn(events::Event) -> std::pin::Pin<std::boxed::Box<(dyn futures::Future<Output = ()> + std::marker::Send + 'static)>> + std::marker::Send + 'static)` cannot be shared between threads safely
   --> src/supervisor/mod.rs:471:27
    |
471 |         let join_handle = task::spawn(run_supervisor_monitor(
    |                           ^^^^^^^^^^^ `(dyn std::ops::Fn(events::Event) -> std::pin::Pin<std::boxed::Box<(dyn futures::Future<Output = ()> + std::marker::Send + 'static)>> + std::marker::Send + 'static)` cannot be shared between threads safely
    | 
   ::: /home/roman/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.22/src/task/spawn.rs:128:20
    |
128 |         T::Output: Send + 'static,
    |                    ---- required by this bound in `tokio::spawn`
    |
    = help: the trait `std::marker::Sync` is not implemented for `(dyn std::ops::Fn(events::Event) -> std::pin::Pin<std::boxed::Box<(dyn futures::Future<Output = ()> + std::marker::Send + 'static)>> + std::marker::Send + 'static)`
```

Reply:
I was able to remove it from `Worker`, `EventNotifier` needs it